pkg/v6/resource/chartconfig/desired.go

```diff
@@ -61,8 +61,8 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		desiredChartConfigs = append(desiredChartConfigs, chartConfig)
 	}
 
-	// Only enable Ingress Controller for Azure.
-	if r.provider == label.ProviderAzure {
+	// Enable Ingress Controller for Azure and AWS.
+	if r.provider == label.ProviderAzure || r.provider == label.ProviderAWS {
```

Comment:
Here the chartconfig is enabled for AWS. The conditional is messy and I want to refactor to use the approach we used for configmaps. So we have resource under each controller with shared logic in a service under pkg.

See https://github.com/giantswarm/giantswarm/issues/4040 for tracking.